src/preview.tsx

```diff
@@ -13,14 +13,14 @@ interface Props {
 export class PreviewForm extends React.PureComponent<Props> {
   public static general: boolean = false; //NOSONAR need to reassign this variable , so can't be readonly
   public static isToc: boolean = false; //NOSONAR need to reassign this variable , so can't be readonly
-  public static isCitation: boolean = false;//NOSONAR need to reassign this variable , so can't be readonly
-  public static isTitle: boolean = false;//NOSONAR need to reassign this variable , so can't be readonly
-  public static tocHeader = [];//NOSONAR need to reassign this variable , so can't be readonly
+  public static isCitation: boolean = false; //NOSONAR need to reassign this variable , so can't be readonly
+  public static isTitle: boolean = false; //NOSONAR need to reassign this variable , so can't be readonly
+  public static tocHeader = []; //NOSONAR need to reassign this variable , so can't be readonly
```

Comment:
Why are these static instead of part of the component state?

Reply:
Tried it using it as a part of the component state, but the values were not getting changed when an option is selected(ex. TOC) and had to stick to static.

Reply:
I can let this slide in this PR, but you haft to remove the NOSONAR comments. Static state makes the component fragile, and this will need to be fixed sooner than later.